metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -882,6 +882,7 @@ class MetalsLanguageServer(
     focusedDocument = Some(path)
     openedFiles.add(path)
     syntheticsDecorator.publishSynthetics(path)
+    worksheetProvider.onDidFocus(path)
```

Comment:
How come this one is necessary? Upon leaving and entering, the `didFocusTextDocument` triggers, which you have below and that makes sense. However, when you first open a document, that one is already working. So how come this change?

Reply:
I think having it in `didOpen` might be needed in case of editors that don't support focusedDocument I think?

Reply:
So I just tested it by turning off `didFocusTextDocument` in nvim-metals and then opening a worksheet. It still evaluates as expected. From my understanding this works currently and having this in here may actually trigger two evaluations on a `didOpen`. Thinking about this more I'm almost positive this used to work, so I'm very curious what happened. The whole point of the `didFocusTextDocument` was because we wanted some way to still trigger this when it was open before and then regained focus. So afaik, I don't think this is needed in `didOpen` since the worksheet will be evaluated no matter what on the first open without it.

Reply:
You're right, I removed it from `didOpen`. I think I misunderstood how `didOpen` might work in other editors. Thanks for pointing this out! I am also not sure why it stopped working, tried to look around, but nothing recent broke this behaviour.

Reply:
ha, it's fresh on my mind since I recently had to think about it for adding worksheet support in nvim-metals.